dragonfly/implementations/psyllid_provider.py

```diff
@@ -0,0 +1,279 @@
+        logger.info('Status of channels: {}'.format(self.status_value_dict))
+        logger.info('Set central frequencies: {}'.format(self.freq_dict))
+        logger.info('Streaming or triggered mode: {}'.format(self.mode_dict))
+
+
+    def request_status(self, channel):
+        logger.info('Checking Psyllid status of channel {}'.format(channel))
+        try:
+            result = self.provider.get(self.queue_dict[channel]+'.daq-status', timeout=10)
+        except core.exceptions.DriplineError:
+            logger.warning('Psyllid instance for channel {} is not running or returned error'.format(channel))
+            self.status_dict[channel]=None
+            self.status_value_dict[channel]=None
+            return self.status_value_dict[channel]
+        except Exception as e:
+            logger.error('Something else went wrong')
+            raise
+        else:
+            self.status_dict[channel] = result['server']['status']
+            self.status_value_dict[channel] = result['server']['status-value']
+            logger.info('Psyllid is running. Status is {}'.format(self.status_dict[channel]))
+            logger.info('Status in numbers: {}'.format(self.status_value_dict[channel]))
+            return self.status_value_dict[channel]
+
+
+
+    def activate(self, channel):
+        if self.status_value_dict[channel] == 0:
+            logger.info('Activating Psyllid instance for channel {}'.format(channel))
+            result = self.provider.cmd(self.queue_dict[channel], 'activate-daq')
+        else:
+            logger.warning('Cannot activate Psyllid instance of channel {}'.format(channel))
+            return False
+        time.sleep(1)
+        self.request_status(channel)
+        if self.status_value_dict[channel]!=4:
+            logger.warning('Activating failed')
+            return False
+        else: return True
+
+
+    def deactivate(self, channel):
+        if self.status_value_dict[channel] != 0:
+            logger.info('Deactivating Psyllid instance of channel {}'.format(channel))
+            result = self.provider.cmd(self.queue_dict[channel],'deactivate-daq')
+        else:
+            logger.warning('Cannot deactivate Psyllid instance of channel {}'.format(channel))
+            return False
+        time.sleep(1)
+        self.request_status(channel)
+        if self.status_value_dict[channel]!=0:
+            logger.warning('Deactivating failed')
+            return False
+        else: return True
+
+
+    def reactivate(self, channel):
+        if self.status_value_dict[channel] == 4:
+            logger.info('Reactivating Psyllid instance of channel {}'.format(channel))
+            result = self.provider.cmd(self.queue_dict[channel], 'reactivate-daq')
+            logger.info(result)
+        else:
+            logger.warning('Cannot reactivate Psyllid instance of channel {}'.format(channel))
+            return False
+        time.sleep(1)
+        self.request_status(channel)
```

Comment:
same questions from activate, note that you did the else block in the preferred way here